sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py

```diff
@@ -501,7 +510,6 @@ def set_read_only(
         :type read_only: bool
         :keyword match_condition: The match condition to use upon the etag
         :paramtype match_condition: :class:`~azure.core.MatchConditions`
-        :keyword str etag: check if the ConfigurationSetting is changed. Set None to skip checking etag
```

Comment:
Why was this removed?

Reply:
Because we are using `configuration_setting.etag` instead of `etag` keyword to check "if_match" and "if_non_match".

Reply:
I don't understand. Can you give more details?

Reply:
It looks like this was a bug - or at least, we were documenting behaviour that we weren't actually doing.
Though etag should be supported as an optional parameter as well. This would overwrite whatever is in the object.

Reply:
Added the line back and track the bug in this issue: https://github.com/Azure/azure-sdk-for-python/issues/30528